e2e/opentelemetry/opentelemetry.test.ts

```diff
@@ -0,0 +1,364 @@
+    ).toBe(0);
+  });
+
+  it('should report http failures', async () => {
+    const serviceName = 'mesh-e2e-test-4';
+    const { fetchJson } = await serve({
+      supergraph,
+      env: {
+        OTLP_EXPORTER_URL: `http://localhost:${jaeger.port}/v1/traces`,
+        OTLP_SERVICE_NAME: serviceName,
+      },
+    });
+
+    await fetchJson('/non-existing').catch(() => {});
+    await sleep(8000);
+    const traces = await getJaegerTraces(serviceName);
+    expect(traces.data.length).toBe(2);
+    const relevantTrace = traces.data.find(trace =>
+      trace.spans.some(span => span.operationName === 'GET /non-existing'),
+    );
+    expect(relevantTrace).toBeDefined();
+    expect(relevantTrace?.spans.length).toBe(1);
+
+    expect(relevantTrace?.spans).toContainEqual(
+      expect.objectContaining({
+        operationName: 'GET /non-existing',
+        tags: expect.arrayContaining([
+          expect.objectContaining({
+            key: 'otel.status_code',
+            value: 'ERROR',
+          }),
+          expect.objectContaining({
+            key: 'error',
+            value: true,
+          }),
+          expect.objectContaining({
+            key: 'http.status_code',
+            value: 404,
+          }),
+        ]),
+      }),
+    );
+  });
+
+  it('context propagation should work correctly', async () => {
+    const traceId = '0af7651916cd43dd8448eb211c80319c';
+    const serviceName = 'mesh-e2e-test-5';
+    const { execute, fetchJson } = await serve({
+      supergraph,
+      env: {
+        OTLP_EXPORTER_URL: `http://localhost:${jaeger.port}/v1/traces`,
+        OTLP_SERVICE_NAME: serviceName,
+      },
+    });
+
+    await expect(
+      execute({
+        query: TEST_QUERY,
+        headers: {
+          traceparent: `00-${traceId}-b7ad6b7169203331-01`,
+        },
+      }),
+    ).resolves.toMatchSnapshot();
+
+    const upstreamHttpCalls = await fetchJson<
+      Array<{
+        url: string;
+        headers?: Record<string, string>;
+      }>
+    >('/upstream-fetch');
+
+    await sleep(8000);
+    const traces = await getJaegerTraces(serviceName);
+    expect(traces.data.length).toBe(3);
+
+    const relevantTrace = traces.data.find(trace =>
+      trace.spans.some(span => span.operationName === 'POST /graphql'),
+    );
+    expect(relevantTrace).toBeDefined();
+
+    // Check for extraction of the otel context
+    expect(relevantTrace.traceID).toBe(traceId);
+    for (const span of relevantTrace.spans) {
+      expect(span.traceID).toBe(traceId);
+    }
+
+    expect(upstreamHttpCalls.length).toBe(8);
+
+    for (const call of upstreamHttpCalls) {
+      const transparentHeader = (call.headers || {})['traceparent'];
+      expect(transparentHeader).toBeDefined();
+      expect(transparentHeader.length).toBeGreaterThan(1);
+      expect(transparentHeader).toContain(traceId);
+    }
+  });
+});
+
+jest.setTimeout(1000000);
```

Comment:
This gets applied to all tests, but is hidden away. If you want to increase the timeout, please do it in the single place for e2es - `tenv.ts`.

Reply:
yeah it was only for testing in development, sorry about that 😆 
will remove